### PR TITLE
Use NetworkClient to implement `sbt -bsp`

### PR DESCRIPTION
**Pull request by eatkins**
Network client already supports the -bsp command (since
65ab7c94d038955d48eefbf7951fb9cd399ab7ef). This commit reworks the
BspClient.run method so that it delegates to the NetworkClient. The
advantage to doing it this way is that improvements to starting up the
sbt server by the thin client will automatically propagate to the -bsp
command. The way that it is implemented, all of the output generated
during server startup will be redirected to System.err which is useful
for debugging without messing up the bsp protocol, which relies on only
bsp messages being written to System.out.

### CODE REVIEW

**Comment by adpi2** [comment]
This is great, thank you!

Unfortunately it does not seem to work on my machine:
1. In a `1.4.3-SNAPSHOT` project
2. Run `sbt -bsp`

The server starts, it loads the project but then it automatically shuts down.

After some investigation I see that the `sbt-console0-ui-trhead` of the server sends an `exit` command to the fast track.

**Comment by eatkins** [comment]
Hmm. It works fine on my computer. It shouldn't make a difference, but what happens when you run `java -jar ~/.ivy2/local/org.scala-sbt/sbt-launch/1.4.3-SNAPSHOT/jars/sbt-launch.jar -bsp`?

**Comment by adpi2** [comment]
Same result
```
> java -jar ~/.ivy2/local/org.scala-sbt/sbt-launch/1.4.3-SNAPSHOT/jars/sbt-launch.jar -bsp
[info] server was not detected. starting an instance
Content-Length: 58
Content-Type: application/vscode-jsonrpc; charset=utf-8

{"jsonrpc":"2.0","method":"shutdown","params":[true,null]}
```

**Comment by eatkins** [comment]
Can you try again with the latest code? Also the version will now be 1.5.0-SNAPSHOT, not 1.4.3-SNAPHOT.

File: main-command/src/main/scala/sbt/internal/client/BspClient.scala
Comment (adpi2) [inline main-command/src/main/scala/sbt/internal/client/BspClient.scala]: `class BspClient` is not used anymore.

File: main-command/src/main/scala/sbt/internal/client/NetworkClient.scala
Comment (adpi2) [inline main-command/src/main/scala/sbt/internal/client/NetworkClient.scala:1223]: `configuration` is not used

File: main-command/src/main/scala/sbt/internal/client/NetworkClient.scala
Comment (adpi2) [inline main-command/src/main/scala/sbt/internal/client/NetworkClient.scala:1220]: `configuration` is not used

**Review by adpi2 — DISMISSED** [review]
It is much better! Also I tested the integration with Metals and it works.

File: main-command/src/main/scala/sbt/internal/client/BspClient.scala
Comment (eatkins) [reply main-command/src/main/scala/sbt/internal/client/BspClient.scala]: Good catch.

File: main-command/src/main/scala/sbt/internal/client/NetworkClient.scala
Comment (eatkins) [reply main-command/src/main/scala/sbt/internal/client/NetworkClient.scala:1220]: Good catch. It is supposed to be used: https://github.com/sbt/sbt/compare/dd2ed16389e1a2c4d082dc6d627b86c8553a432b..f96eeb9b09a9a94b1b5c079fc52497053dd592ae

**Comment by eatkins** [comment]
As a side note, `sbt --client -bsp` works as well using sbtn. The one catch is that we need a new sbtn that redirects the server startup output to System.err rather than System.out, which is a part of this PR. Besides the lower startup time, `sbtn -bsp` uses about 15MB of RSS memory compared to 350MB for `sbt -bsp`.

**Comment by eatkins** [comment]
@adpi2 I simplified the approach for ensuring that the server doesn't unexpectedly exit. Can you verify that it works on your machine? Also, I need a re-approval due to force pushing.

**Review by adpi2 — APPROVED** [review]

**Comment by adpi2** [comment]
Apparently, my approving review is not enough because I don't have the write access on this repository.

**Comment by eed3si9n** [comment]
@adpi2 Welcome to the team!

**Pull request merged**
